crates/revm/src/executor.rs

```diff
@@ -431,35 +431,48 @@ pub fn commit_state_changes<DB>(
                 Entry::Vacant(entry) => {
                     let entry = entry.insert(Default::default());
                     entry.info = account.info.clone();
-
-                    let account = to_reth_acc(&entry.info);
-                    if !(has_state_clear_eip && account.is_empty()) {
-                        post_state.create_account(block_number, address, account);
+                    entry.account_state = AccountState::NotExisting; // we will promote account state down the road
+                    let new_account = to_reth_acc(&entry.info);
+
+                    #[allow(clippy::nonminimal_bool)]
+                    // If account was touched before state clear EIP, create it.
+                    if !has_state_clear_eip ||
+                        // If account was touched after state clear EIP, create it only if it is not empty.
+                        (has_state_clear_eip && !new_account.is_empty())
```

Comment:
If before state clear, create account.
If after state clear and account is empty, do nothing.

After state clear EIP, only returned empty account from db are the ones that were touched (and are marked for removal) so I a not sure if this is triggered but either way check is good.